src/_actions/TradeActions.js

```diff
@@ -171,6 +171,17 @@ export const updatePurchaseInfo = (index, fieldName, fieldValue) => ({
     fieldValue,
 });
 
+// TODO: this is arbritary, need discuss how we should handle it
+const trySubscribeUntilSuccess = c =>
+    LiveData.api
+        .subscribeToOpenContract(c)
+        .then(response => {
+            const contract = response.proposal_open_contract;
+            if (contract.validation_error && Object.keys(contract).length < 3) {
+                setTimeout(() => trySubscribeUntilSuccess(c), 300);
+            }
+        });
```

Comment:
@borisyankov should we even retry ?
Backend might fail for all kind of reason that frontend does not have information about, do you think keep on retry make sense?